src/cpp/core/include/core/http/Response.hpp

```diff
@@ -287,7 +289,8 @@ class Response : public Message
       // send the file from its stream
       try
       {
-         return setBody(*pIfs, filter, buffSize, padding);
+         bool precompressed = boost::algorithm::ends_with(filePath.getAbsolutePath(), ".gz");
```

Comment:
Should this also check for files ending in `.br`, or are we only ever requesting `.gz` files?

Reply:
This is an awkward consequence of how the current `Response` class methods call one another. Further down the chain `setBody()` will automatically gzip anything with a `content-encoding: gzip` header, so we need to add an escape hatch to avoid that for files that are already compressed -- we can't add the header afterwards. But we don't have an automatic Brotli step in `setBody()` (preferring Gzip for all cases), so I didn't need to add an escape hatch for it. A better variable name here would be `alreadyGzipped` rather than `precompressed`, I suppose.